lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -95,6 +98,20 @@ def success?(response)
   end
 
   def error_message(response)
-    JSON.parse(response.body)['error']['full_messages'].join(' ') rescue "unknown error (response #{response.code})"
+    messages = {
+      '400' => '400 Bad Request: Something is wrong with the data we sent to Foreman server',
+      '401' => '401 Unauthorized Request: Check credentials for validity',
+      '404' => '404 Not Found: The requested resource was not found',
```

Comment:
This feels like an edge case to me. The only thing I can think of is that they entered the wrong server URL and end up on a different service.

Reply:
It's definitely an edge case but I had a few more in mind when I wrote it, based on situations I've seen previously:

1. Attempted improper use of load balancing

2. "I forgot I cloned this VM and left it running on the same network"

3. Automation/scripts run amok